### PR TITLE
Add random backoff algorithm to delay between re-runs

### PR DESCRIPTION
**Pull request by andrea-segalini**
The option "reruns-delay-backoff" makes the delay specified with "reruns-delay" vary according to a random backoff algorithm.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -71,6 +71,15 @@ test re-run is launched:
 
    $ pytest --reruns 5 --reruns-delay 1
 
+To use a random backoff to vary the delay, ``--reruns-delay-backoff`` command.
```
Comment (cjriches) [inline README.rst]: "use the ``--reruns-delay-backoff`` command line option"

for consistency with other documentation

File: README.rst [hunk 2]
```diff
@@ -71,6 +71,15 @@ test re-run is launched:
...
+This causes the delay to be picked randomly between 0 and a maximum value that
+double at every fail (starting value is set by ``--reruns-delay``):
```
Comment (srowe) [inline README.rst]: "doubles on every failure"

File: pytest_rerunfailures.py [hunk 3]
```diff
@@ -186,13 +195,24 @@ def get_reruns_delay(item):
     rerun_marker = _get_marker(item)
 
     if rerun_marker is not None:
+        args = copy(rerun_marker.args).pop(0)
```
Comment (cjriches) [inline pytest_rerunfailures.py]: This being python, I'm trying to follow the types and getting very confused. As far as I can tell, `rerun_marker.args` is a tuple, and this doesn't have a `pop` method. `pop` would also return the first argument, which makes no sense for a variable called `args`.

Comment (cjriches) [reply pytest_rerunfailures.py]: Also, you seem to be treating the `copy` module as a function here which looks wrong (i.e. you did `import copy` and not `from copy import copy`.

File: pytest_rerunfailures.py [hunk 4]
```diff
@@ -186,13 +195,24 @@ def get_reruns_delay(item):
...
+
         if "reruns_delay" in rerun_marker.kwargs:
             delay = rerun_marker.kwargs["reruns_delay"]
-        elif len(rerun_marker.args) > 1:
-            # check for arguments
-            delay = rerun_marker.args[1]
+        elif len(args):
```
Comment (cjriches) [inline pytest_rerunfailures.py]: `len(args)` is redundant, can just do `elif args` for non-empty I believe

File: README.rst (hunk 2 above)
Comment (andrea-segalini) [reply README.rst]: Done

File: README.rst (hunk 1 above)
Comment (andrea-segalini) [reply README.rst]: Done

File: pytest_rerunfailures.py (hunk 3 above)
Comment (andrea-segalini) [reply pytest_rerunfailures.py]: Sorry I didn't test that bit at all :(

Fixed

File: pytest_rerunfailures.py (hunk 4 above)
Comment (andrea-segalini) [reply pytest_rerunfailures.py]: Dropped this bit, I'm using the same strategy as the existing code for this.

**Comment by andrea-segalini** [comment]
Rebased on-top of release 10.2, as they bumped the min Python version supported to 3.7 after that.

**Pull request closed without merging**